lib/src/flutter/reporters/slack_reporter.dart

```diff
@@ -0,0 +1,226 @@
+  /// The number of scenario failures before the test suite will terminate
+  /// and cease reporting to Slack. Defaults to `15`.
+  final int maximumToleratedFailures;
+
+  final SlackMessenger slackMessenger;
+
+  /// Reported as the initial message (or parent thread when [threadResults] is `true`)
+  /// of the test run. Defaults to `Starting tests`.
+  final String startLabel;
+
+  /// Whether the test run should be threaded under a single message. Defaults `true`.
+  final bool threadResults;
+
+  final features = <FinishedMessage>[];
+  final scenarios = <ScenarioFinishedMessage>[];
+  final scenariosInActiveFeature = <ScenarioFinishedMessage>[];
+  StepFinishedMessage firstFailedStepInActiveScenario;
+
+  SlackReporter(
+    this.slackMessenger, {
+    this.maximumToleratedFailures = 15,
+    this.threadResults = true,
+    this.startLabel = 'Starting tests',
+  });
+
+  @override
+  Future<void> onException(exception, stackTrace) {
+    final payload = [
+      slackMessenger.divider,
+      slackMessenger.buildTextSection(':bangbang: Exception :bangbang:'),
+      slackMessenger.buildTextSection(exception.toString()),
+      slackMessenger.buildTextSection('```$stackTrace```'),
+      slackMessenger.divider,
+    ];
+
+    return slackMessenger.notifySlack(payload);
+  }
+
+  @override
+  Future<void> message(msg, level) async {
+    if (level == MessageLevel.error) await slackMessenger.sendText(msg);
+  }
+
+  @override
+  Future<void> onFeatureFinished(feature) async {
+    features.add(feature);
+    final allScenariosPassed = scenariosInActiveFeature.every((s) => s.passed);
+
+    if (allScenariosPassed) {
+      await slackMessenger.sendText(
+          ':white_check_mark: ${feature.name} (${scenariosInActiveFeature.length}/${scenariosInActiveFeature.length} passed)');
+    } else {
+      final passedScenarios = scenariosInActiveFeature.where((s) => s.passed);
+      await slackMessenger.sendText(
+          ':warning: ${feature.name} (${passedScenarios.length}/${scenariosInActiveFeature.length} passed)');
+    }
+
+    scenariosInActiveFeature.clear();
+  }
+
+  @override
+  Future<void> onScenarioFinished(scenario) async {
+    scenarios.add(scenario);
+    scenariosInActiveFeature.add(scenario);
+    if (!scenario.passed) {
+      final payload = [
+        slackMessenger.buildTextSection(':x: ${scenario.name}'),
+        slackMessenger.buildTextSection(
+            'Failed at step: ${firstFailedStepInActiveScenario.name}'),
+      ];
+      await slackMessenger
+          .notifySlack(payload)
+          .then((_) => firstFailedStepInActiveScenario = null);
+    }
+
+    if (scenarios.where((s) => !s.passed).length > maximumToleratedFailures) {
+      await slackMessenger.sendText(
+          ':x: :x: :x: :x: Aborting: too many failures :x: :x: :x: :x:');
+      exit(1);
+    }
+  }
+
+  @override
+  Future<void> onStepFinished(step) async {
+    if (step.result.result != StepExecutionResult.pass &&
```

Comment:
@tshedor It would be good here to check if there are any attachments (such as failed step images) and use the upload method to Slack?  You can see the code in the JSONReporter on how to achieve this.  Obviously you would only need to do this if the attachment had an image mime type.

See https://github.com/jonsamwell/dart_gherkin/blob/master/lib/src/reporters/json/json_step.dart#L54